rust/src/proofs/types.rs

```diff
@@ -230,6 +253,37 @@ impl Drop for fil_VanillaProof {
     }
 }
 
+#[repr(C)]
+pub struct fil_AggregateProof {
+    pub status_code: FCPResponseStatus,
+    pub error_msg: *const libc::c_char,
+    pub proof_len: libc::size_t,
+    pub proof_ptr: *const u8,
+}
+
+impl Default for fil_AggregateProof {
+    fn default() -> fil_AggregateProof {
+        fil_AggregateProof {
+            status_code: FCPResponseStatus::FCPNoError,
+            error_msg: ptr::null(),
+            proof_len: 0,
+            proof_ptr: ptr::null(),
+        }
+    }
+}
+
+impl Drop for fil_AggregateProof {
+    fn drop(&mut self) {
```

Comment:
I think this leaks the `error_msg` if there is one

Reply:
It's not allocated/managed, it just points to a rust string from the ffi toolkit as far as I know (similar to all other usages of the `error_msg`)

Reply:
If you implement `Drop` manually, you also need to drop `error_msg`. In other structs the `#[derive(DropStructMacro)]` implements `Drop`. Though in this case you can just use the `DropStructMacro` derive instead (see https://github.com/filecoin-project/rust-fil-ffi-toolkit/tree/master/drop-struct-macro-derive#readme for more information).

Reply:
Thanks, I see, will take a closer look.  At one point, I may have needed to impl Drop rather than use the macro, but not sure that's the case any more.